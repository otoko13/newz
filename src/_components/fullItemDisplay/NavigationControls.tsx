import React from 'react';
import { Stack } from 'office-ui-fabric-react/lib/Stack';
import { IconButton } from 'office-ui-fabric-react/lib/Button';
import './navigationControls.scss';

export interface INavigationControlsProps {	
    onNextClick: () => void;
    onPreviousClick: () => void;
    hasNext: boolean;
    hasPrevious: boolean;	
}	

const NavigationControls = (props: INavigationControlsProps) => {	
    return (
        <Stack horizontal horizontalAlign='space-between' className='NavigationControls'>
            <Stack.Item>
                {props.hasPrevious &&
                    <IconButton aria-label='Previous item' className='navigation-button' iconProps={{iconName: 'ChevronLeft'}} onClick={props.onPreviousClick} />
                }
            </Stack.Item>	
            <Stack.Item>
                {props.hasNext &&
                    <IconButton aria-label='Next item' className='navigation-button' iconProps={{iconName: 'ChevronRight'}} onClick={props.onNextClick} />
                }               
            </Stack.Item>
        </Stack>	
    );	
}	

export default NavigationControls; 