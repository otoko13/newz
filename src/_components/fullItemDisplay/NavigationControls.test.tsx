import { act, render } from '@testing-library/react';
import * as React from 'react';
import NavigationControls, { INavigationControlsProps } from './NavigationControls';

let props: INavigationControlsProps;

beforeEach(() => {
    props = {
        hasNext: true,
        hasPrevious: true,
        onNextClick: jest.fn(),
        onPreviousClick: jest.fn(),
    }
});

describe('snapshot', () => {
    it('should look sensible', async () => {
        const { container } = render(<NavigationControls {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});

describe('showing buttons', () => {
    it('should not show a please select something message when there is an item', async () => {
    });

    it('should show the item instead if there is one', async () => {
    });
});
