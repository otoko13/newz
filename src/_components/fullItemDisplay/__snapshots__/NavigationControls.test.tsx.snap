// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should look sensible 1`] = `
<div
  class="ms-Stack NavigationControls css-40"
>
  <div
    class="ms-StackItem css-41"
  >
    <button
      aria-label="Previous item"
      class="ms-Button ms-Button--icon navigation-button root-42"
      data-is-focusable="true"
      type="button"
    >
      <span
        class="ms-Button-flexContainer flexContainer-43"
        data-automationid="splitbuttonprimary"
      >
        <i
          aria-hidden="true"
          class="ms-Icon root-33 ms-Button-icon icon-45"
          data-icon-name="ChevronLeft"
          role="presentation"
        />
      </span>
    </button>
  </div>
  <div
    class="ms-StackItem css-41"
  >
    <button
      aria-label="Next item"
      class="ms-Button ms-Button--icon navigation-button root-42"
      data-is-focusable="true"
      type="button"
    >
      <span
        class="ms-Button-flexContainer flexContainer-43"
        data-automationid="splitbuttonprimary"
      >
        <i
          aria-hidden="true"
          class="ms-Icon root-33 ms-Button-icon icon-45"
          data-icon-name="ChevronRight"
          role="presentation"
        />
      </span>
    </button>
  </div>
</div>
`;
